paimon-api/src/main/java/org/apache/paimon/rest/HttpClient.java

```diff
@@ -19,89 +19,87 @@
 import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.core.JsonProcessingException;
 
-import okhttp3.Headers;
-import okhttp3.HttpUrl;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
+import org.apache.hc.client5.http.classic.methods.HttpDelete;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.classic.methods.HttpUriRequestBase;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManagerBuilder;
+import org.apache.hc.client5.http.io.HttpClientConnectionManager;
+import org.apache.hc.client5.http.ssl.DefaultClientTlsStrategy;
+import org.apache.hc.client5.http.ssl.HttpsSupport;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.message.BasicHeader;
+import org.apache.hc.core5.net.URIBuilder;
+import org.apache.hc.core5.reactor.ssl.SSLBufferMode;
+import org.apache.hc.core5.ssl.SSLContexts;
 
+import java.io.IOException;
+import java.net.URISyntaxException;
 import java.nio.charset.StandardCharsets;
-import java.time.Duration;
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.Map;
 import java.util.function.Function;
 
-import static okhttp3.ConnectionSpec.CLEARTEXT;
-import static okhttp3.ConnectionSpec.COMPATIBLE_TLS;
-import static okhttp3.ConnectionSpec.MODERN_TLS;
-import static org.apache.paimon.rest.LoggingInterceptor.DEFAULT_REQUEST_ID;
-import static org.apache.paimon.rest.LoggingInterceptor.REQUEST_ID_KEY;
-
-/** HTTP client for REST catalog. */
+/** Apache HTTP client for REST catalog. */
 public class HttpClient implements RESTClient {
 
-    private static final OkHttpClient HTTP_CLIENT =
-            new OkHttpClient.Builder()
-                    .retryOnConnectionFailure(true)
-                    .connectionSpecs(Arrays.asList(MODERN_TLS, COMPATIBLE_TLS, CLEARTEXT))
-                    .addInterceptor(new ExponentialHttpRetryInterceptor(5))
-                    .addInterceptor(new LoggingInterceptor())
-                    .connectTimeout(Duration.ofMinutes(3))
-                    .readTimeout(Duration.ofMinutes(3))
-                    .writeTimeout(Duration.ofMinutes(3))
-                    .build();
-
-    private static final MediaType MEDIA_TYPE = MediaType.parse("application/json");
+    private static volatile CloseableHttpClient httpClient;
```

Comment:
Why creating httpClient lazily? Will creating a client start a thread, or is it just lazy